Src/DependencyCollector/Shared/Implementation/HttpDesktopDiagnosticSourceSubscriber.cs

```diff
@@ -47,7 +52,20 @@ public void OnNext(DiagnosticListener value)
             {
                 if (value.Name == "System.Net.Http.Desktop")
                 {
-                    this.sourceSubscription = value.Subscribe(this.parent, (Predicate<string>)null);
+                    this.sourceSubscription = value.Subscribe(
+                        this.parent, 
+                        (evnt, r, _) =>
+                        {
+                            if (r != null && evnt == "System.Net.Http.Desktop.HttpRequestOut")
+                            {
+                                // request is never null
+                                var request = (HttpWebRequest)r;
+                                return !this.applicationInsightsUrlFilter.IsApplicationInsightsUrl(request.RequestUri.ToString());
```

Comment:
Since the performance test overrides the profile API and quick pulse endpoint - this change will not help much specifically to performance tests. It will help customers for sure.

I'd suggest an easy fix in `IsApplicationInsightsUrl` - do the check `if (SdkInternalOperationsMonitor.IsEntered())` before making url comparisons. Once you'd implement this fix - call to `ToString()` here will be unnecessary